email_notification/admin.py

```diff
@@ -84,7 +90,70 @@ def clean_html_content_sv(self):
 
 
 @admin.register(EmailTemplate)
-class EmailTemplateAdmin(admin.ModelAdmin):
+class EmailTemplateAdmin(ExtraButtonsMixin, admin.ModelAdmin):
     model = EmailTemplate
     form = EmailTemplateAdminForm
     exclude = ["html_content"]
+
+    @button(label="Email Template Testing")
+    def template_tester(self, request):
```

Comment:
I thought about this and ended up separating `EmailNotificationContext` into a separate file. I also created a static `from_form` function there so that `admin.py` stays cleaner. This way this whole context thingie and stuff related to it are wrapped into one class.